src/PreparedPostTypes.php

```diff
@@ -62,10 +70,22 @@ protected function setPostTypes(array $postTypes)
 	 */
 	public function restBaseToSlug(string $restBase)
 	{
-		if (in_array($restBase, $this->getPostTypeRestBases())) {
+		if ($this->has($restBase)) {
 			return $this->postTypes[$restBase];
 		}
 
 		return null;
 	}
+
+	/**
+	 * Checks if the given rest_base exists for the prepared post types.
+	 *
+	 * @param string $restBase
+	 *
+	 * @return bool
+	 */
+	public function has(string $restBase) : bool
```

Comment:
This method naming matches the PSR for containers. This object contains prepared post types. If it also gained a get method, it could implement psr-11. Not sure if that's valuable though.

https://www.php-fig.org/psr/psr-11/#112-reading-from-a-container

Reply:
Right, this class is a container for the post types that have a rest base.  It stores those post types as well as the rest bases.  Therefore, "prepared" really means " a post type with a rest base."

Using the `Psr\Container\ContainerInterface`, you can utilize both the `get` and `has` methods.  The problem with the `get` though is this:  **are you getting the post types or the rest bases?**

Think about it.  This container holds 2 different sets of data: post types that have a rest base AND an array of rest bases from those post types.

If we walk through the _intent_ of this class, it's been named as a container for "prepared" post types.  Think about when you want to get a value out of this container, what value are you wanting to go fetch: post type or rest base?

- If you decide "rest base", then rename the class to `RestBaseContainer`.
- If you decide "post type", then I'd advocate changing the class name to `PostTypeContainer` or `RestBasePostTypeContainer`.